test/artemis/stores/mapgraph_store_test.cljs

```diff
@@ -692,12 +691,84 @@
                 :numberField 3
                 :nullField   nil
                 :nestedObj   nil
-                ::cache      "root"}}}})
+                ::cache      "root"}}}
+
+     :union-array
+     {:query    (d/parse-document
+                  "{
+                     search(text: \"a\") {
+                       ... on object {
+                         id
+                         stringField
+                         __typename
+                       }
+                       ... on otherobject {
+                         id
+                         numberField
+                         __typename
+                       }
+                     }
+                   }")
+      :result   {:search
+                 [{:id          "abcd"
+                   :stringField "this is a string"
+                   :__typename  "object"}
+                  {:id          "efgh"
+                   :numberField 3
+                   :__typename  "otherobject"}]}
```

Comment:
Will this omit an object from the search array that isn't of type "object" and "otherobject"? Going to dig into all this a bit later when I can fire up a repl

Reply:
@haywoood if i'm understanding you correctly you're asking: if an object returned in the search array doesn't have a corresponding entry in `id-attrs` for it's type, will mapgraph normalize it properly? (same question would apply if the object returned doesn't have the proper unique field for it's type, `:id` in this case)

good q, maybe we should add tests for `union-array-no-id` and `nested-union-array-no-id`

Reply:
Yeah, will add tests for non-id'd unions.